mautic/mautic
# Cleanup exported reports

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | N
| New feature/enhancement? (use the a.x branch)      | Y
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | Y 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
Exported csv or zip reports should be deleted after sending it in email attachment. For too big files that can't be attached we are sending a link to a zip file and that file should be deleted after 7 days.


#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Export a report that will be sent in an attachment as a csv file. (will happen if csv file size <= defined max size)
3. Export a report that is big and will be sent in an attachment as a zip file. (will happen if csv file size  > defined max size but after zipping csv, zip file size is <= defined max size)
4. Export a report that is too big and will be sent as a link for download a zip file. (will happen if zip file size > defined max size)
5. First two files csv and zip file will be deleted right after the email is sent. Third zip file will deleted after 7 days.

**use `report_export_max_filesize_in_bytes` config param to define file size limit.**
